cobc/codegen.c

```diff
@@ -2763,8 +2762,28 @@ output_source_cache (void)
 
 /* Literal */
 
+static void
+cb_cache_program_decimal_constant (struct cb_program *prog, struct literal_list *cached_literal)
+{
+	struct literal_list	*l;
+	if (prog != NULL && cached_literal->make_decimal) {
```

Comment:
Is `prog` ever `NULL` here?
Should we only call that function if `make_decimal` is true (=checked outside)?

In any case, please add a function doc comment for new functions (and wherever you work on a function without, feel invited to add one).

Reply:
Well, if `prog` is NULL, it's actually a bug. Should I just let the compiler crash in this case ?

Performing the check on `make_decimal` inside the function looks safer to me (one does not have to think whether he should or shouldn't call the function) ; of course, as long as it is only ever used in `cb_lookup_literal`, this is not a big deal...

Reply:
yes, please let the compiler SIGSEGV there, and if this happens in `make checkall` then there's something related to fix

as per your comment - as this is only a function with local state - let's do the check outside

Reply:
No SIGSEGV, we're all good ;)